syntax = "proto3";

option go_package = "/calculatorpb";

// Membuat service Calculator untuk server dan client
service Calculator {
    rpc Add(AddRequest) returns (AddReply) {}
    rpc Subtract(SubtractRequest) returns (SubtractReply) {}
    rpc Multiply(MultiplyRequest) returns (MultiplyReply) {}
    rpc Divide(DivideRequest) returns (DivideReply) {}
}

// Request Add
message AddRequest {
    int32 n1 = 1;
    int32 n2 = 2;
}

// Reply Add
message AddReply {
    int32 n1 = 1;
}

// Request Substract
message SubtractRequest {
    int32 n1 = 1;
    int32 n2 = 2;
}

// Reply Substract
message SubtractReply {
    int32 n1 = 1;
}

// Request Multiply
message MultiplyRequest {
    int32 n1 = 1;
    int32 n2 = 2;
}

// Reply Multiply
message MultiplyReply {
    int32 n1 = 1;
}

// Request Divide
message DivideRequest {
    int32 n1 = 1;
    int32 n2 = 2;
}

// Reply Divide
message DivideReply {
    float n1 = 1;
}